docs/news/posts/2024-11-15-aiidalab-mvc.md

```diff
@@ -0,0 +1,322 @@
+        return response.json().get("uuid", "")
+```
+
+```python
+
+from app.backends import Backend
+
+class AiiDAService:
+    def __init__(self, backend: Backend):
+        self._backend = backend
+
+    def get_process_uuid(self):
+        return self._backend.fetch_process_uuid()
+```
+
+In the above example, the `AiiDAService` is injected with a concrete backend depending on the needs of the app. For example, one could design an app that is seeded with (or auto-detects) information regarding the local environment. If the app is deployed on the same server as AiiDA, it could communicate with AiiDA directly by leveraging a `DbBackend`. If it is instead deployed on a server separated from the AiiDA instance, it could leverage a `RestBackend` to communicate with AiiDA over HTTP. Note that in any case, the API of the `AiiDAService` is agnostic to the concrete `Backend` instance - the service simply makes a call to `fetch_process_uuid`, which is a required method of _any_ `Backend`. Thus, maintenance of the `Service` class is reduced through logic encapsulation in the backend classes.
+
+**Tip 💡** for those interested, we encourage further reading on the above principles to better understand the pros and cons of the suggested patterns.
+
+## Optimization
+
+Lastly, it is often the case that some operations in a given AiiDAlab app may take longer than others. To ensure a smooth user experience (UX), it is crucial to take the following into consideration.
+
+### Lazy loading
+
+The entirety of the app is not required at any given moment. Consider implementing lazy loading techniques to limit the impact of rendering on the user. For example, a lazy-loaded view could be designed as follows:
+
+```python
+
+class View(ipw.VBox):
+
+    def __init__(self, **kwargs):
+        super().__init__(
+            children=[
+                LoadingWidget("Loading the view"),  # or any custom spinner widget
+                **kwargs,
+            ]
+        )
+
+        self.rendered = False
+
+    def render(self):
+        if self.rendered:
+            return
+
+        self.dropdown = ipw.Dropdown(description="Options")
+        self.textbox = ipw.Text(description="Label")
+        self.button = ipw.Button(
+            description="Submit",
+            button_style="success",
+            icon="check",
+        )
+        children=[
+            self.dropdown,
+            ipw.HBox(
+                children=[
+                    self.textbox,
+                    self.button,
+                ],
+            ),
+        ]
+
+        self.rendered = True
+```
+
+An associated `Controller` can handle the rendering of the `View` (calling `render()`) at the precise moment it is visited by the user, thus deferring the loading impact to a later stage of the app.
+
+**Tip 💡** to learn more, you can check out [this PR](https://github.com/aiidalab/aiidalab-qe/pull/802) implementing lazy loading in the Quantum ESPRESSO app.
```

Comment:
"Also, feel free to leave a review, as no-one in the `aiidateam` wants to go through 10k lines of changed code" :D
Wouldn't call this "Tip", though. Maybe instead add a footnote?

Reply:
Right, footnote would be better. Also, while we're on the subject, feel free to review the PR :)